taichi/ir/type_factory.cpp

```diff
@@ -103,6 +103,20 @@ PrimitiveType *TypeFactory::get_primitive_int_type(int bits, bool is_signed) {
   return int_type->cast<PrimitiveType>();
 }
 
+PrimitiveType *TypeFactory::get_primitive_real_type(int bits) {
+  Type *real_type;
+  if (bits == 16) {
+    real_type = get_primitive_type(PrimitiveTypeID::f16);
+  } else if (bits == 32) {
+    real_type = get_primitive_type(PrimitiveTypeID::f32);
+  } else if (bits == 64) {
+    real_type = get_primitive_type(PrimitiveTypeID::f64);
+  } else {
+    TI_ERROR("No primitive real type has {} bits", bits);
+  }
+  return real_type->cast<PrimitiveType>();
+}
```

Comment:
Not really useful in this PR. Maybe we can save it for the future, or I can remove it.

Reply:
Saving it here is fine.